loader/icd.c

```diff
@@ -189,6 +191,127 @@ void khrIcdVendorAdd(const char *libraryName)
     }
 }
 
+void khrIcdLayerAdd(const char *libraryName)
+{
+    void *library = NULL;
+    cl_int result = CL_SUCCESS;
+    pfn_clGetLayerInfo p_clGetLayerInfo = NULL;
+    pfn_clInitLayer p_clInitLayer = NULL;
+    struct KHRLayer *layerIterator = NULL;
+    struct KHRLayer *layer = NULL;
+    cl_layer_api_version api_version = 0;
+    const struct _cl_icd_dispatch *targetDispatch = NULL;
+    const struct _cl_icd_dispatch *layerDispatch = NULL;
+    cl_uint layerDispatchNumEntries = 0;
+    cl_uint loaderDispatchNumEntries = 0;
+
+    // require that the library name be valid
+    if (!libraryName)
+    {
+        goto Done;
+    }
+    KHR_ICD_TRACE("attempting to add vendor %s...\n", libraryName);
```

Comment:
There are a few minor copy-paste bugs in this function.

vendor --> layer

clIcdGetPlatformIDsKHR --> clInitLayer

Reply:
Thanks Ben, nice catches.